slack-lint/src/main/java/slack/lint/text/SpanPointMarkDangerousCheckDetector.kt

```diff
@@ -0,0 +1,122 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package slack.lint.text
+
+import com.android.tools.lint.client.api.UElementHandler
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.LintFix
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import com.android.tools.lint.detector.api.UastLintUtils
+import org.jetbrains.uast.UBinaryExpression
+import org.jetbrains.uast.UExpression
+import org.jetbrains.uast.UReferenceExpression
+import org.jetbrains.uast.UastBinaryOperator
+import org.jetbrains.uast.tryResolve
+import slack.lint.text.SpanPointMarkDangerousCheckDetector.Companion.ISSUE
+import slack.lint.util.sourceImplementation
+
+/**
+ * Checks for SpanPointMarkDangerousCheck. See [ISSUE].
+ */
+class SpanPointMarkDangerousCheckDetector : Detector(), SourceCodeScanner {
+
+  companion object {
+    private fun Implementation.toIssue() = Issue.create(
+      id = "SpanPointMarkDangerousCheck",
+      briefDescription = "Check Span Flags Using Bitmask",
+      explanation =
+      "Spans flags can have priority or other bits set. " +
+        "Ensure that Span flags are checked using " +
+        "`currentFlag and Spanned.SPAN_POINT_MARK_MASK == desiredFlag` " +
+        "rather than just `currentFlag == desiredFlag`",
+      category = Category.CORRECTNESS,
+      priority = 4,
+      severity = Severity.ERROR,
+      implementation = this
+    )
+
+    val ISSUE = sourceImplementation<SpanPointMarkDangerousCheckDetector>().toIssue()
+  }
+
+  override fun getApplicableUastTypes() = listOf(UBinaryExpression::class.java)
+
+  override fun createUastHandler(context: JavaContext): UElementHandler {
+    return ReportingHandler(context)
+  }
+
+  /**
+   * Reports violations of SpanPointMarkDangerousCheck.
+   */
+  private class ReportingHandler(private val context: JavaContext) : UElementHandler() {
+    companion object {
+      private const val SPANNED_CLASS = "android.text.Spanned"
+      private val markPointFields = setOf(
+        "$SPANNED_CLASS.SPAN_INCLUSIVE_INCLUSIVE",
+        "$SPANNED_CLASS.SPAN_INCLUSIVE_EXCLUSIVE",
+        "$SPANNED_CLASS.SPAN_EXCLUSIVE_INCLUSIVE",
+        "$SPANNED_CLASS.SPAN_EXCLUSIVE_EXCLUSIVE",
+      )
+      private const val MASK_CLASS = "$SPANNED_CLASS.SPAN_POINT_MARK_MASK"
+    }
+
+    override fun visitBinaryExpression(node: UBinaryExpression) {
+      if (node.operator is UastBinaryOperator.ComparisonOperator) {
```

Comment:
Would it be more accurate to check `UastBinaryOperator.EQUALS` (or not equals) as a part of this?